### ISSUE TITLE
Camera Preview In Editor Causes Performance Drop

### ISSUE THREAD

**Issue by ScoutWozniak** [Facepunch/sbox-issues#4644]
### Describe the bug

Whenever the camera preview is shown, performance decreases massively.

### To Reproduce

1. Add a camera to the scene
2. Make sure the preview is open

### Expected behavior

Shouldn't cause performance to drop

### Media/Files

https://github.com/Facepunch/sbox-issues/assets/99102549/0146e168-a907-4dcb-9788-b093a16027d9

**Comment by garrynewman** [comment]
You're meant to have a camera preview tool pop up - do you see that anywhere?

**Comment by ScoutWozniak** [comment]
It's there for me, I guess it just didn't show up in the recording.

**Comment by handsomematt** [comment]
Can you post your hardware specs? It's likely a problem with Vulkan but I can't start debugging any of this stuff properly without narrowing down what devices it affects

**Comment by ScoutWozniak** [comment]
I'm using a Ryzen 7 3700X, RTX 3060 on Windows 10 with 16 gigs of RAM.  Need anything else?

**Comment by handsomematt** [comment]
I think I've got it on a similar machine now, is your console getting spammed with anything?

**Comment by garrynewman** [comment]
This might be related to the thing I just added, where we wait manually for the pixmap to get written

**Comment by MrBrax** [comment]
it's rendering the scene twice, isn't this what should happen? been like that for me since it was added, didn't think anything of it

**Comment by Jerichos** [comment]
I had similar issue few days ago when I opened s&box and created an empty project. Scene view had 1fps (game view completely fine when switched tab). When I get home I will check it again and update comment.

**Comment by Walrusking16** [comment]
https://github.com/Facepunch/sbox-issues/assets/20746458/0f00d25f-ae67-4e04-970f-847f5296d9b3
I took a video of it happening, it is kind of hard to tell the fps drop on the video, the tools performance jumps up to 30ms when the camera is selected. I also included the verbose console output, and used the following specs:

128gb ram, 3080ti, ryzen 9 5950x

**Comment by Jerichos** [comment]
Tested today again and there was still significant frame drop in scene view.
**Resolved by updating graphics driver.**

**Comment by ScoutWozniak** [comment]
Graphics drivers are up to date now but after testing it again I'm still experiencing the lag.  I'll attach more footage of the editor and the log file (the camera preview doesn't seem to show up in obs but it is still there in editor).

https://github.com/Facepunch/sbox-issues/assets/99102549/4da5e7d2-bb37-4db6-87d0-9acce402ee98

[Log.log](https://github.com/Facepunch/sbox-issues/files/14110320/Log.log)

**Comment by MrBrax** [comment]
![1706704380492](https://github.com/Facepunch/sbox-issues/assets/1517911/4468c813-d776-4d0d-bb0d-e365c7a6e669)

i think the driver version check needs some more logic

**Comment by MrBrax** [comment]
if you have a still supported gpu that is, a nag for something i can't solve doesn't really help

**Comment by MrBrax** [comment]
the dialog box above that opens every time i start the game. i can't do anything about it so it's just a nag. the game still works on this machine in a pinch.

**Comment by cnaro1996** [comment]
![image](https://github.com/Facepunch/sbox-issues/assets/12787159/ea6c2601-6645-43f1-8893-c64b926a80d5)

i'm also having this problem now.

running:
ryzen 9 5900x
RTX 3080
32 gigs of RAM
windows 10

running latest drivers but also was having the issue before updating
![image](https://github.com/Facepunch/sbox-issues/assets/12787159/3288a88f-29ee-491e-baf2-20dd61da64c7)

any ideas of possible cause? everything is quite smooth except for during the camera preview. then it drops to something like 2-3 fps in the scene.

**Comment by Dickarus666** [comment]
![image](https://github.com/Facepunch/sbox-issues/assets/12787159/ea6c2601-6645-43f1-8893-c64b926a80d5)

i'm also having this problem now.

running:
ryzen 9 5900x
RTX 3080
32 gigs of RAM
windows 10

running latest drivers but also was having the issue before updating
![image](https://github.com/Facepunch/sbox-issues/assets/12787159/3288a88f-29ee-491e-baf2-20dd61da64c7)

any ideas of possible cause? everything is quite smooth except for during the camera preview. then it drops to something like 2-3 fps in the scene.

**Comment by Dickarus666** [comment]
![image](https://github.com/Facepunch/sbox-issues/assets/159097837/0826ccde-8658-419f-a910-a92918c25351)
i'm also having this problem now.

running:
ryzen 9 5900x
RTX 3080
32 gigs of RAM
windows 10

running latest drivers but also was having the issue before updating
![image](https://github.com/Facepunch/sbox-issues/assets/159097837/ddb7d820-a556-4610-9e4b-bc57ef4de81e)

log: [Log.log](https://github.com/Facepunch/sbox-issues/files/14321072/Log.log)

any ideas of possible cause? everything is quite smooth except for during the camera preview. then it drops to something like 2-3 fps in the scene.

**Comment by garrynewman** [comment]
This has been happening for a month now @samzanemesis  - any ideas? Did you look into it?

**Comment by Dickarus666** [comment]
reminder bump

**Comment by Dickarus666** [comment]
wanted to post an update, this problem might not be related to the s&box editor as much as it relates to local system graphics configurations. today i ran into trouble with my vulkan drivers on a separate game, and ended up using wagnard's display driver uninstaller to do a deep removal of driver settings in an attempt to fix the vulkan issues. after doing this and then manually reinstalling the newest graphics drivers, [this seems to inadvertently helped the s&box camera preview issue as well.](https://www.youtube.com/watch?v=KdvQj09WKuI)



it's not entirely fixed, but it's gone from 2-3 fps to just a stutter every few seconds now, which is much much better. i'm suspecting the remaining stutter might just be from some further nvidia configuration needed on my part.

tl;dr: corrupted vulkan graphics drivers may have been the issue. a deep cleaning with DDU followed by a reinstall has (almost) fixed the problem.
